From: Miao Xie <miaox@cn.fujitsu.com>
Date: Tue, 22 Jan 2013 10:50:01 +0000
Patch-mainline: 3.8
Git-commit: eebc60840636e7351371fc17bcd057384bf0c16a
References: FATE#312888
Subject: [PATCH] Btrfs: check the return value of
 btrfs_run_ordered_operations()

We forget to check the return value of btrfs_run_ordered_operations() when
flushing all the pending stuffs, fix it.

Signed-off-by: Miao Xie <miaox@cn.fujitsu.com>
Signed-off-by: Josef Bacik <jbacik@fusionio.com>
Signed-off-by: David Sterba <dsterba@suse.cz>
---
 fs/btrfs/transaction.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

--- a/fs/btrfs/transaction.c
+++ b/fs/btrfs/transaction.c
@@ -1449,9 +1449,9 @@ static int btrfs_flush_all_pending_stuff
 	 * it here and no for sure that nothing new will be added
 	 * to the list
 	 */
-	btrfs_run_ordered_operations(root, 1);
+	ret = btrfs_run_ordered_operations(root, 1);
 
-	return 0;
+	return ret;
 }
 
 /*
